### PR TITLE
Adds a Still class

### PR DESCRIPTION
**Pull request by azyth**
compiles and integrates a still class to the firmware package, 

may need a little tweaking once the PID functionality is completed. but for now should not bother anything else, so should be safe to merge.

### CODE REVIEW

**Comment by azyth** [comment]
@kyleoliveira take a look see what you think

File: CMakeLists.txt [hunk 1]
```diff
@@ -23,6 +23,8 @@ set(SOURCE_FILES
     ../lib/Ohmbrewer_Temperature.cpp
     ../lib/Ohmbrewer_RIMS.h
     ../lib/Ohmbrewer_RIMS.cpp
+    ../lib/Ohmbrewer_STILL.cpp
```
Comment (kyleoliveira) [inline CMakeLists.txt]: 'Still' isn't an acronym, so you want to Titlecase this (i.e. 'Still').

You're gonna need to do this in a lot of places, of course.

File: lib/Ohmbrewer_STILL.cpp [hunk 2]
```diff
@@ -0,0 +1,558 @@
+//
+// Created by azyth on 10/12/15.
+//
```
Comment (kyleoliveira) [inline lib/Ohmbrewer_STILL.cpp]: You're going to want to clear out this comment up here.

File: lib/Ohmbrewer_STILL.cpp [hunk 3]
```diff
@@ -0,0 +1,558 @@
...
+ * @param elementPins[ heating powerPin ; heating controlPin ]
+ * @param tempBus - onewire temperature pin
+ */
+Ohmbrewer::STILL::STILL(int id, int (&elementPins)[2], int tempBus ) : Ohmbrewer::Equipment(id) {
+
+
+    //Thermostat
+    //Temp therm constructor with no array checking
+    int thermPins[3] = {tempBus};
+    memcpy(&thermPins+sizeof(int), elementPins, 2*sizeof(int));
```
Comment (kyleoliveira) [inline lib/Ohmbrewer_STILL.cpp]: Why not do something like

``` c++
int thermPins[3] = {tempBus, elementPins[0], elementPins[1]}
```

? Just curious.

File: lib/Ohmbrewer_STILL.cpp [hunk 4]
```diff
@@ -0,0 +1,558 @@
...
+
+    return start - millis();
+}
+
+/**
+ * The Equipment state. True => On, False => Off
+ * @returns True => On, False => Off
+ */
+bool Ohmbrewer::STILL::getState() const {
+    return getCoolantPump()->getState() || getKettleSensor()->getState() || getTherm()->getState();
```
Comment (kyleoliveira) [inline lib/Ohmbrewer_STILL.cpp]: I think you're missing some parts here, right?

File: lib/Ohmbrewer_STILL.cpp [hunk 5]
```diff
@@ -0,0 +1,558 @@
...
+/**
+ * Reports which of the Rhizome's pins are occupied by the
+ * Equipment, forming a logical Sprout.
+ * @param pins The list of physical pins that the Equipment is connected to.
+ */
+void Ohmbrewer::STILL::whichPins(std::list<int>* pins){
+
+    pins->push_back(_kettleSensor->getBusPin()); //assumes onewire protocol
+    _therm->whichPins(pins);
+    _coolant->whichPins(pins);
```
Comment (kyleoliveira) [inline lib/Ohmbrewer_STILL.cpp]: You need to push_back the results for all of these (2 out of 3 aren't getting pushed in)

File: lib/Ohmbrewer_STILL.h [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+         */
+        TemperatureSensor* _columnSensor;
+        /**
+         * The temperature sensor in the condenser
+         */
+        TemperatureSensor* _coolantSensor;
+        /**
+         * The recirculation pump for the condenser
+         */
+        Pump* _coolant;
```
Comment (kyleoliveira) [inline lib/Ohmbrewer_STILL.h]: Probably should be `_coolantPump` for to be explicit.

File: lib/Ohmbrewer_STILL.cpp (hunk 5 above)
Comment (azyth) [inline lib/Ohmbrewer_STILL.cpp]: yes they are, this is passing pins to those whichpin functions witch is pushing them on itself.

File: firmware/rhizome.ino [hunk 7]
```diff
@@ -104,13 +105,13 @@ void loop() {
...
 //                                                                                                  ->getState()); // The Thermostat's HeatingElement
-        ((Ohmbrewer::RIMS*)sprouts.front())->getTube()
+        ((Ohmbrewer::Still*)sprouts.front())->getTherm()
                                            ->getElement()
                                            ->toggleState();
 //        navi->publish(); // (*)
         lastUpdate = millis();
     }
-    ((Ohmbrewer::RIMS*)sprouts.front())->getTube()->getSensor()->doWork(); //Temp patch to make sensor read in TUBE
+    ((Ohmbrewer::Still*)sprouts.front())->getColumnSensor()->doWork(); //Temp patch to make sensor read in TUBE
```
Comment (kyleoliveira) [inline firmware/rhizome.ino]: This should call work() instead of doWork(). Not a showstopper, but just sayin'.

Comment (azyth) [inline firmware/rhizome.ino]: getColumnSensor->work()???? there is no work function , is there?

File: lib/Ohmbrewer_Still.cpp [hunk 8]
```diff
@@ -0,0 +1,558 @@
...
+Ohmbrewer::Still::Still(int id, int (&elementPins)[2], int tempBus ) : Ohmbrewer::Equipment(id) {
+
+
+    //Thermostat
+    //Temp therm constructor with no array checking
+    int thermPins[3] = {tempBus, elementPins[0], elementPins[1]};
+//    memcpy(&thermPins+sizeof(int), elementPins, 2*sizeof(int));
+//    _therm = new Thermostat(id+4, thermPins);
+
+    _therm = new Thermostat(id+4, thermPins );
```
Comment (kyleoliveira) [inline lib/Ohmbrewer_Still.cpp]: Would this be
```c++
_therm = new Thermostat(id+4, &thermPins);
```
?

File: firmware/rhizome.ino (hunk 7 above)
Comment (kyleoliveira) [inline firmware/rhizome.ino]: There is... it's in Equipment.

**Comment by azyth** [comment]
Holding on this till after the beta release. when i am in bed i will probably try to get this working.

**Comment by kyleoliveira** [comment]
Gonna close this for now. A lot of stuff has changed since it was opened up, so you're going to at least need to rebase off of master if nothing else.

**Pull request closed without merging**